prov/efa/src/rxr/rxr_init.c

```diff
@@ -875,5 +875,12 @@ EFA_INI
 	if (rxr_env.enable_shm_transfer && rxr_get_local_gids())
 		return NULL;
 
+	if (getenv("FI_EFA_SHM_MAX_MEDIUM_SIZE")) {
+		FI_WARN(&rxr_prov, FI_LOG_FABRIC,
```

Comment:
I do not think this what we want. This means if user has the FI_EFA_SHM_MAX_MEDIUM_SIZE environment variable. EFA will not be picked and by default the user does not see a warning.

So what will likely to happen is user end up using tcp, and got worse performance.

I think what we want is "fprintf(stderr, xxx)" followed by "abort()"

Reply:
Yep, makes total sense, thank you. I've made those changes but also moved the code around a bit to be with the other "FI_EFA_*" environment variable checks.